saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/OpenSamlAuthenticationProvider.java

```diff
@@ -474,6 +480,33 @@ else if (subject.getEncryptedID() != null) {
 		return username;
 	}
 
+	private Map<String, Object> getAssertionAttributes(Assertion assertion) {
+		Map<String, Object> attributeMap = new HashMap<>();
```

Comment:
Let's please use a `LinkedHashMap` so that it preserves the order in which OpenSAML gave them to us.